src/components/Simulator/utils.ts

```diff
@@ -15,3 +17,8 @@ export const shareOnTwitter = (): void => {
 export const clearUrlParams = (location: Location): void => {
   navigate(location.pathname, { replace: true })
 }
+
+export const getValidPathId = (pathIdString: PathId | null): PathId | null => {
```

Comment:
This could be a good case to use [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) where you receive any string and you return a more specific type, in this case `PathId`

```ts
export const getValidPathId = (pathIdString: string | null): PathId | null => {
  if (!isValidPathId(pathIdString)) return null
  return pathIdString
}

const isValidPathId = (pathIdString: string | null): pathIdString is PathId => {
  return PATH_IDS.includes(pathIdString as PathId)
}
```